File: app/src/main/java/com/example/cs388_project_3_flixster/MainActivity.java
package com.example.cs388_project_3_flixster;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.bumptech.glide.Glide;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private RequestQueue requestQueue;
    private List<Movie> movieList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerview);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        requestQueue = VolleySingleton.getmInstance(this).getRequestQueue();

        movieList = new ArrayList<>();

        MovieAdapter adapter = new MovieAdapter(MainActivity.this, movieList);
        recyclerView.setAdapter(adapter);

        requestQueue = VolleySingleton.getmInstance(this).getRequestQueue();
        fetchMovies();

    }

    private void fetchMovies() {
        String url = "https://api.themoviedb.org/3/movie/now_playing?api_key=a07e22bc18f5cb106bfe4cc1f83ad8ed&language=en-US&page=1";
        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                        try {
                            JSONArray array = new JSONObject(response).getJSONArray("results");
                            for (int i = 0; i < array.length(); i++) {

                                JSONObject jsonObject = array.getJSONObject(i);
                                String title = jsonObject.getString("original_title");
                                String overview = jsonObject.getString("overview");
                                String posterPath = jsonObject.getString("poster_path");
                                Double rating = jsonObject.getDouble("vote_average");

                                Movie movie = new Movie(title, posterPath, overview, rating);
                                movieList.add(movie);

                            }
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                        MovieAdapter adapter = new MovieAdapter(MainActivity.this, movieList);
                        recyclerView.setAdapter(adapter);
                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                // Log the error for debugging
                Log.e("VolleyError", "Error: " + error.getMessage());
            }
        });

        requestQueue.add(stringRequest);
    }
}

File: app/src/main/java/com/example/cs388_project_4_flixsterp2/MainActivity.kt
package com.example.cs388_project_4_flixsterp2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.android.volley.Request
import com.android.volley.VolleyError
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import org.json.JSONException
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    private val movieList: MutableList<Movie> = mutableListOf()
    private val otherMovieList: MutableList<OtherMovie> = mutableListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // API REQUEST for RecyclerViewTop
        val recyclerViewTop = findViewById<RecyclerView>(R.id.recyclerviewTop)
        val layoutManagerTop = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        recyclerViewTop.layoutManager = layoutManagerTop
        val adapterTop = MovieAdapter(this, movieList)
        recyclerViewTop.adapter = adapterTop
        makeApiRequestForRecyclerViewTop(adapterTop)

        // API REQUEST for RecyclerViewBottom
        val recyclerViewBottom = findViewById<RecyclerView>(R.id.recyclerviewBottom)
        val layoutManagerBottom = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        recyclerViewBottom.layoutManager = layoutManagerBottom
        val adapterBottom = MovieAdapterBottom(this, otherMovieList)
        recyclerViewBottom.adapter = adapterBottom
        makeApiRequestForRecyclerViewBottom(adapterBottom)
    }

    private fun makeApiRequestForRecyclerViewBottom(adapter: MovieAdapterBottom) {
        // Instantiate the RequestQueue.
        val queue = Volley.newRequestQueue(this)
        val url = "https://api.themoviedb.org/3/discover/movie?api_key=a07e22bc18f5cb106bfe4cc1f83ad8ed&language=en-US&page=1"
        // Request a string response from the provided URL.
        val stringRequest = StringRequest(Request.Method.GET, url,
            { response ->
                try {
                    val json = JSONObject(response)
                    val results = json.getJSONArray("results")
                    for (i in 0 until results.length()) {
                        val movieData = results.getJSONObject(i)
                        val originalTitle = movieData.getString("original_title")
                        val rating = movieData.getString("vote_average")
                        val description = movieData.getString("overview")
                        val imageUrl = "https://image.tmdb.org/t/p/original" + movieData.getString("poster_path")
                        val movie = OtherMovie(originalTitle, rating, description, imageUrl)
                        otherMovieList.add(movie)
                    }
                    adapter.notifyDataSetChanged()
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            },
            { error: VolleyError ->
                error.printStackTrace()
            })

        // Add the request to the RequestQueue.
        queue.add(stringRequest )
    }

    private fun makeApiRequestForRecyclerViewTop(adapter: MovieAdapter) {
        // Instantiate the RequestQueue.
        val queue = Volley.newRequestQueue(this)
        val url = "https://api.themoviedb.org/3/discover/tv?api_key=a07e22bc18f5cb106bfe4cc1f83ad8ed&language=en-US&page=1"
        // Request a string response from the provided URL.
        val stringRequest = StringRequest(Request.Method.GET, url,
            { response ->
                try {
                    val json = JSONObject(response)
                    val results = json.getJSONArray("results")
                    for (i in 0 until results.length()) {
                        val movieData = results.getJSONObject(i)
                        val originalTitle = movieData.getString("name")
                        val rating = movieData.getString("vote_average")
                        val imageUrl = "https://image.tmdb.org/t/p/original" + movieData.getString("poster_path")
                        val movie = Movie(originalTitle, rating, imageUrl)
                        movieList.add(movie)
                    }
                    adapter.notifyDataSetChanged()
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            },
            { error: VolleyError ->
                error.printStackTrace()
            })

        // Add the request to the RequestQueue.
        queue.add(stringRequest )
    }

}
